### DISCUSSION TITLE
How to send an Embed?

### DISCUSSION THREAD

**Discussion by DeadLyBro** [rigwild/discord-self-bot-console#6] [Q&A] [answered]
Can you write how to send embed? Let others benefit as well. 

Note: I just edit, nothing new.

**Comment by rigwild** [comment, marked as answer]
EDIT:

**SENDING EMBEDS AS A _USER ACCOUNT_ IS NOT POSSIBLE ANYMORE, DISCORD UPDATED ITS API (see [this reddit post](https://web.archive.org/web/20220209223900/https://www.reddit.com/r/Discord_selfbots/comments/sa0hc2/discord_embeds_patched/))**

---

Looks like the issue is the color, you need to send an integer (see https://discord.com/developers/docs/resources/channel#embed-object-embed-structure).

I found this nice embed generator online. Click `+` at the bottom in the embed section then copy the `embed` key in the JSON output.
https://discord.club/dashboard

I updated the API to accept a full embed (https://github.com/rigwild/discord-self-bot-console/commit/f9d15bd44a72226e92b34aa410c6ce255d10d964).
```js
api = {
  sendEmbed: (channelId, embed = { title: 'Title', description: 'Description' }) => apiCall(`/channels/${channelId}/messages`, { embed }, 'POST')
}
```

Full example:

```ts
id()

const embed = {
  title: 'Title',
  color: 5022192,
  description: 'Description',
  timestamp: '2021-12-09T13:32:00.000Z',
  url: 'https://url.url/',
  author: {
    name: 'Author',
    url: 'https://author.url/',
    icon_url: 'https://picsum.photos/200'
  },
  image: {
    url: 'https://picsum.photos/200'
  },
  thumbnail: {
    url: 'https://picsum.photos/50'
  },
  footer: {
    text: 'Footer',
    icon_url: 'https://picsum.photos/20'
  },
  fields: [
    {
      name: 'Field1_key',
      value: 'Field1_value'
    },
    {
      name: 'Field2_key',
      value: 'Field2_value',
      inline: true
    }
  ]
}

await api.sendEmbed(cid, embed)
// or
await api.apiCall(`/channels/${cid}/messages`, { embed }, 'POST')
```

Output:

![embed example](https://user-images.githubusercontent.com/26366184/145406696-8734410a-fc08-40e3-abf8-de12d8d89db9.png)

**Reply by DeadLyBro** [reply]
Thanks for your answer, I hope it helps people who don't understand.

**Reply by DeadLyBro** [reply]
@rigwild Dude u must delete old syntax from Readme.md 

![image](https://image.prntscr.com/image/8AEL9WdJQ6iXnVZnwjnxeg.png)

**Reply by rigwild** [reply]
> @rigwild Dude u must delete old syntax from Readme.md

Good catch :) thanks

**Reply by bariscodefxy** [reply]
Ohh, this is a very old answer

**Comment by bariscodefxy** [comment]
Yes, Discord updated its API but still you can send embeds. How? Let's explain. I found a embed generator that can give an customizable embed url. When u send it to Discord u can see your embed message.

![Screenshot_20221122_000145](https://user-images.githubusercontent.com/85716242/203157518-dbda1ad8-c3e4-4e0b-9106-651521ee848f.jpg)

The Generator Website: https://embedl.ink/

You can customize your embed with GET parameters like this: https://embedl.ink/?deg&provider=&providerurl=&author=&authorurl=&title=&color=%23000000&media=none&mediaurl=&desc=

**Reply by rigwild** [reply]
Clever! Thanks for sharing.

**Reply by DeadLyBro** [reply]
Nice but if we don't have permission to send Links? :/

**Reply by bariscodefxy** [reply]
> Nice but if we don't have permission to send Links? :/

You can't do it without permission. After Discord API update, there is one way to send embeds, sending links.
